src/amo/components/HeroRecommendation/index.js

```diff
@@ -66,58 +75,76 @@ export class HeroRecommendationBase extends React.Component<InternalProps> {
     let heading;
     let link;
 
-    const linkProps = _checkInternalURL({
-      urlString: this.makeCallToActionURL(),
-    }).isInternal
-      ? {}
-      : { rel: 'noopener noreferrer', target: '_blank' };
+    const heightClassName =
+      siteIsReadOnly || siteNotice
+        ? 'HeroRecommendation--height-with-notice'
+        : 'HeroRecommendation--height-without-notice';
 
-    if (addon) {
-      heading = addon.name;
-      link = (
-        <Link
-          className="HeroRecommendation-link"
-          onClick={this.onHeroClick}
-          to={this.makeCallToActionURL()}
-        >
-          {linkInsides}
-        </Link>
-      );
-    } else if (external) {
-      heading = external.name;
-      link = (
-        <a
-          className="HeroRecommendation-link"
-          href={this.makeCallToActionURL()}
-          onClick={this.onHeroClick}
-          {...linkProps}
-        >
-          {linkInsides}
-        </a>
+    if (shelfData) {
+      gradientsClassName = `HeroRecommendation-${gradient.start}-${gradient.end}`;
+      log.info(
+        `className ${gradientsClassName} generated from the API response. This should match a selector in styles.scss`,
       );
-    }
 
-    // translators: If uppercase does not work in your locale, change it to lowercase.
-    // This is used as a secondary heading.
-    const recommended = i18n.gettext('RECOMMENDED');
-    const gradientsClassName = `HeroRecommendation-${gradient.start}-${gradient.end}`;
-    log.info(
-      `className ${gradientsClassName} generated from the API response. This should match a selector in styles.scss`,
-    );
+      const linkInsides = <span> {i18n.gettext('Get the extension')} </span>;
+      const linkProps = _checkInternalURL({
+        urlString: this.makeCallToActionURL(),
+      }).isInternal
+        ? {}
+        : { rel: 'noopener noreferrer', target: '_blank' };
+
+      if (addon) {
+        heading = addon.name;
+        link = (
+          <Link
+            className="HeroRecommendation-link"
+            onClick={this.onHeroClick}
+            to={this.makeCallToActionURL()}
+          >
+            {linkInsides}
+          </Link>
+        );
+      } else if (external) {
+        heading = external.name;
+        link = (
+          <a
+            className="HeroRecommendation-link"
+            href={this.makeCallToActionURL()}
+            onClick={this.onHeroClick}
+            {...linkProps}
+          >
+            {linkInsides}
+          </a>
+        );
+      }
+    } else {
+      gradientsClassName = `HeroRecommendation--loading`;
+    }
 
     return (
       <section
-        className={makeClassName('HeroRecommendation', gradientsClassName, {
-          'HeroRecommendation--no-image': !featuredImage,
-        })}
+        className={makeClassName(
+          'HeroRecommendation',
+          gradientsClassName,
+          heightClassName,
+          {
+            'HeroRecommendation--no-image': !featuredImage,
```

Comment:
The layout without an image will be uncommon so the loading screen should follow the featured image layout instead.  The image skeleton image could probably just be a square styled `<LoadingText>` (a pulsing gray square).

Reply:
I wasn't sure which was going to be more common, an image or no image, so I went with no image for now. I'll check with @devaneymoz to see which he thinks will be more common and I'll file a follow-up to change it if necessary.

Reply:
@devaneymoz says (from Slack):

> images will be used very rarely (think: major releases like FPN, not so much individual rando extensions)

Therefore I think leaving the skeleton as the non-image version makes sense.

Reply:
Oh! I had no idea that the image layout will be uncommon. Man, that's a bummer. I am not a fan of the image-less layout.